org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/TokenTypeMapper.java

```diff
@@ -42,6 +44,9 @@ public IToken apply(final String tokenType) {
 				return tokenProvider.getToken(tokenType);
 			}
 		}
-		return TMUIPlugin.getThemeManager().getDefaultTheme().getToken(tokenType);
+		final ITheme[] defaultTheme = new ITheme[1];
+		// Needs to be executed in UI-thread:

```

Comment:
Wasn't it fixed in TM4E directly?

Reply:
Yes, but I've seen that LSP4E is using the TM4E release for building.

Reply:
I think it would be OK to move LSP4E to use latest snapshot of TM4E until its next release. What do you think?

Reply:
I agree.

Reply:
I was planning to do a release once https://github.com/eclipse/lsp4e/pull/935 is merged for simrel. If you agree, I think we should wait for that before using TM4E snapshot.

Reply:
Sorry, I've seen your comment tool late. It's fine for me. I'll provide a new commit...